streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1038,85 +1044,11 @@ boolean maybeCommit() {
     private void maybeUpdateStandbyTasks() {
         if (state == State.RUNNING && taskManager.hasStandbyRunningTasks()) {
             if (processStandbyRecords) {
```

Comment:
The original motivation I think, is a very "weird" heuristic, the current thread host a standby task A` while its corresponding active task A is likely hosted by another thread, if that thread does not process A and hence write more records to the changelog, then we would not get any for A` to update -- furthermore, if the changelog is a source topic, if the other thread hosting A does not commit, then we cannot update A` any further because of limit offset. 

And when this thread is committing for another active task, say B, then it is "likely" that the other thread has committed for its active task A too, so it is a good time to try to update A`.

Like I said, this is not a very good one as an after-though, when we (@vvcephei) work on the thread class itself, we can discuss whether to keep this or not.

Reply:
But `processStandbyRecords` is a thread-level variable, and different threads aren't really synchronized on when they commit within an instance much less across instances (and the threads would have to be from different instances since we would never host standby A and active A together).  Granted the commit interval should be the same, but won't the actual commit times drift out of sync?
Also, even if this heuristic does "work" and saves us from trying to process a standby when it's already caught up and the active hasn't yet committed new records, if a standby is at all behind the active task it seems like this "optimization" will only make it fall further behind. But, we can revisit this during the thread class refactoring

Reply:
Yes I agree with you, I feel this heuristic does not really worth it and there maybe better ones that we can use to replace it.

Reply:
Woah. That's a pretty subtle heuristic. It would definitely not hurt to clean it up, too.